# HG changeset patch
# User Kai Habel <kai.habel@gmx.de>
# Date 1296733625 -3600
#      Thu Feb 03 12:47:05 2011 +0100
# Branch release-3-4-x
# Node ID 3458c5bba6e58dd2f303de0c0299531e59c24765
# Parent  4f4eb7da82d56b0013e63e2006b60e6873122355
use text_renderer object only when freetye is available

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2011-02-03  Kai Habel  <kai.habel@gmx.de>
+
+	* src/gl-render.cc(text_to_pixels): Use text_renderer object only
+	when freetype is available
+
 2011-02-03  John W. Eaton  <jwe@octave.org>
 
 	* ov-base.cc (octave_base_value::as_mxArray): Return 0 silently.
 	* mex.cc (mxArray_octave_value::dup): If value can't be
 	converted to mxArray object, then simply create a clone of the
 	mxArray_octave_value container.
 
 2010-02-02  Rik  <octave@nomad.inbox5.com>
diff --git a/src/gl-render.cc b/src/gl-render.cc
--- a/src/gl-render.cc
+++ b/src/gl-render.cc
@@ -3209,18 +3209,20 @@ opengl_renderer::make_marker_list (const
 }
 
 void
 opengl_renderer::text_to_pixels (const std::string& txt,
                                  uint8NDArray& pixels,
                                  Matrix& bbox,
                                  int halign, int valign, double rotation)
 {
+#if HAVE_FREETYPE
   text_renderer.text_to_pixels (txt, pixels, bbox,
                                 halign, valign, rotation);
+#endif
 }
 
 Matrix
 opengl_renderer::render_text (const std::string& txt,
                             double x, double y, double z,
                             int halign, int valign, double rotation)
 {
 #if HAVE_FREETYPE
